### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by ayo-klaytn**
## Add Client Libraries
ethers.js, web3.js

- Describe your changes to communicate to the maintainers why we should accept this pull request.
- If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

Please put an x in the boxes related to your change.

- [] Minor Issues and Typos
- [x] Major Content Contribution
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to reach out. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn-docs/blob/master/CONTRIBUTING.md)
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn-docs)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large content contribution, kick off the discussion by explaining why you would suggest the content contribution, etc...

### CODE REVIEW

File: docs/dapp/sdk/web3-js.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+In this guide, you'll learn how to use the web3.js library to send a transaction, read data from the blockchain and interact with an existing contract on the Klaytn Network.
+
+
+# Prerequisites
+
+* Code-Editor: a source-code editor such as [VS-Code](https://code.visualstudio.com/download).
+* [Metamask](https://docs.klaytn.foundation/dapp/tutorials/connecting-metamask#install-metamask): used to deploy the contracts, sign transactions and interact with the contracts.
+* RPC Endpoint: you can get this from one of the supported [Endpoint Providers](https://docs.klaytn.foundation/content/dapp/json-rpc/public-en).
+* Test KLAY from [Faucet](https://baobab.wallet.klaytn.foundation/faucet): fund your account with sufficient KLAY.
+* [NodeJS and NPM](https://kinsta.com/blog/how-to-install-node-js/)
```
Comment (praveen-klaytn) [inline docs/dapp/sdk/web3-js.md:16]: Hi @ayo-klaytn ,  
  
I think instead of providing some blog links which may expire later on, its might be better to provide official website like [https://nodejs.org/en/](https://nodejs.org/en/)

File: docs/dapp/sdk/ether-js.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+In this guide, you'll learn how to use the Ethers.js library to send a transaction, read data from the blockchain and interact with an existing contract on the Klaytn Network.
+
+
+# Prerequisites
+
+* Code-Editor: a source-code editor such as [VS-Code](https://code.visualstudio.com/download).
+* [Metamask](https://docs.klaytn.foundation/dapp/tutorials/connecting-metamask#install-metamask): used to deploy the contracts, sign transactions and interact with the contracts.
+* RPC Endpoint: you can get this from one of the supported [Endpoint Providers](https://docs.klaytn.foundation/content/dapp/json-rpc/public-en).
+* Test KLAY from [Faucet](https://baobab.wallet.klaytn.foundation/faucet): fund your account with sufficient KLAY.
+* [NodeJS and NPM](https://kinsta.com/blog/how-to-install-node-js/)
```
Comment (praveen-klaytn) [inline docs/dapp/sdk/ether-js.md:17]: Can provide Nodejs official website links instead of blogs as they we may not have control on them

File: docs/dapp/sdk/web3-js.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+**Output**
+
+To run the script and interact with smart contracts, you can run the following command in your terminal: 
+
+```js
+node interact.js
+```
+
+If the transaction was succesful, in your terminal you'll see the transaction hash and the value stored.
```
Comment (praveen-klaytn) [inline docs/dapp/sdk/web3-js.md:237]: It will be helpful if etherjs documentation links are provided as reference

File: docs/dapp/sdk/ether-js.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+**Output**
+
+To run the script and interact with smart contracts, you can run the following command in your terminal: 
+
+```js
+node interact.js
+```
+
+If the transaction was successful, in your terminal you'll see the transaction hash and the value stored.
```
Comment (praveen-klaytn) [inline docs/dapp/sdk/ether-js.md:231]: It will be helpful if etherjs documentation links are provided as reference

**Review by praveen-klaytn — APPROVED** [review]

File: docs/SUMMARY.md [hunk 5]
```diff
@@ -202,7 +202,7 @@
       * [subbridge](dapp/json-rpc/api-references/subbridge.md)
     * [Transaction Error Codes](dapp/json-rpc/transaction-error-codes.md)
   * [RPC Service Providers](dapp/rpc-service.md)
-  * [Klaytn SDKs](dapp/sdk/README.md)
+  * [Client Libraries](dapp/sdk/README.md)
```
Comment (aidan-kwon) [inline docs/SUMMARY.md:205]: How about this? We usually call caver SDK, so this change may confuse ecosystem developers
```suggestion
  * [Client SDK & Libraries](dapp/sdk/README.md)
```

File: docs/dapp/sdk/ether-js.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+**Output**
+
+To run the script and send data to the blockchain, you can run the following command in your terminal: 
+
+```js
+node send.js
+```
+
+If the transaction was succesful, you'll see the transaction receipt been logged out in your terminal.
```
Comment (toniya-klaytn) [inline docs/dapp/sdk/ether-js.md:146]:
```suggestion
If the transaction was succesful, you'll see the transaction receipt been logged in your terminal.
```

File: docs/getting-started/hardhat/hardhat.md [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+
+    function _burn(uint256 tokenId) internal override(KIP17) {
+        super._burn(tokenId);
+    }
+}
+```
+
+**Code Walkthrough**
+
+This is your smart contract. **line 1** shows that Hardhat uses the Solidity version 0.8.7 or greater. Other than that, it imports KIP7.sol and other supporting contracts. From **lines 6-12**, a smart contract that inherits KIP7 is been created. Also, the token name and symbol was passed in the constructor.
```
Comment (aidan-kwon) [inline docs/getting-started/hardhat/hardhat.md:174]: Not KIP17?

File: docs/SUMMARY.md [hunk 8]
```diff
@@ -31,15 +31,16 @@
...
+  * [Deploying Smart Contract Using KEN](getting-started/quick-start/README.md)
     * [Launch an Endpoint Node](getting-started/quick-start/launch-an-en.md)
     * [Top up your Account](getting-started/quick-start/top-up-your-account.md)
     * [Install Development Tools](getting-started/quick-start/install-development-tools.md)
     * [Deploy a Smart Contract](getting-started/quick-start/deploy-a-smart-contract.md)
     * [Check the Deployment](getting-started/quick-start/check-the-deployment.md)
-  * [Account Management](getting-started/account/README.md)
-    * [Creating Accounts](getting-started/account/creating-accounts.md)
-    * [Managing Accounts](getting-started/account/managing-accounts.md)
+    * [Account Management](getting-started/account/README.md)
```
Comment (aidan-kwon) [inline docs/SUMMARY.md:41]: Is this change intended?

File: docs/getting-started/hardhat/hardhat.md [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+* Explore Hardhat forking feature.
+
+
+# 1. Pre-requisites
+
+To follow this tutorial, the following are the prerequisites:
+
+Code editor: a source-code editor such [VS-Code](https://code.visualstudio.com/download).
+[Metamask](https://docs.klaytn.foundation/dapp/tutorials/connecting-metamask#install-metamask): used to deploy the contracts, sign transactions and interact with the contracts.
+[AllthatNode](https://www.allthatnode.com/): an RPC Node provider.
```
Comment (aidan-kwon) [inline docs/getting-started/hardhat/hardhat.md:27]: Why don't you put the RPC Endpoint page of Klaytn docs like other new pages?

File: docs/dapp/sdk/ether-js.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
+![](./images/Klaytn-ether.js.png)
+
+# Introduction
+
+[ethers.js](https://docs.ethers.org/v5/) is a JavaScript library that allows developers to interact with EVM-compatible blockchain networks like Klaytn. With Klaytn supporting features for [Ethereum Equivalence](https://medium.com/klaytn/using-ethereum-tools-in-klaytn-dc068d48de04), Ethereum tools such as ethers.js can be used on Klaytn without any significant modifications.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:5]: Do you any specific reason you add v5? now it is v6.

File: docs/dapp/sdk/ether-js.md [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+
+Thus, developers can leverage this compatibility and use the Ethers.js library to interact with a Klaytn node.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:7]:
```suggestion
Thus, developers can leverage this compatibility and use the ethers.js library to interact with a Klaytn node.
```

File: docs/dapp/sdk/ether-js.md [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+
+In this guide, you'll learn how to use the Ethers.js library to send a transaction, read data from the blockchain and interact with an existing contract on the Klaytn Network.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:9]:
```suggestion
In this guide, you'll learn how to use the ethers.js library to send a transaction, read data from the blockchain and interact with an existing contract on the Klaytn Network.
```

File: docs/dapp/sdk/ether-js.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+To get started, you need to create a project directory to house the files to be created in this guide.
+
+```js
+mkdir ethers-sample
+cd ethers-sample
+```
+
+## Install ethers.js
+
+To install ethers.js run the following command in your terminal:
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:32]:
```suggestion
To install ethers.js, run the following command in your terminal:
```

File: docs/dapp/sdk/ether-js.md [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+
+```js
+npm install --save ethers
+```
+
+## Initialize ethers.js
+
+In this tutorial, we would be creating a bunch of scripts file to send transactions, read data from the blockchain, and also interact with existing smart contract. To get started, you need to know how to initialize ethers for each of your script files.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:40]:
```suggestion
In this tutorial, we would be creating a bunch of script files to send transactions, read data from the blockchain, and also interact with an existing smart contract. To get started, you need to know how to initialize ethers.js for each of your script files.
```

File: docs/dapp/sdk/ether-js.md [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+After successfully importing ethers, you need to connect to Klaytn by instantiating a new ethers.js `JsonRpcProvider` object with an RPC URL of the Klaytn network. Add the code below to the existing code:
+
+```js
+const url = 'RPC URL';
+const provider = new ethers.JsonRpcProvider(url)
+```
+
+# Reading data from the blockchain
+
+To read data from the blockchain, create a new `read.js` file in your project folder by running this command
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:59]:
```suggestion
To read data from the blockchain, create a new `read.js` file in your project folder by running this command:
```

File: docs/dapp/sdk/ether-js.md [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+* RPC Endpoint: you can get this from one of the supported [Endpoint Providers](https://docs.klaytn.foundation/content/dapp/json-rpc/public-en).
+* Test KLAY from [Faucet](https://baobab.wallet.klaytn.foundation/faucet): fund your account with sufficient KLAY.
+* [NodeJS and NPM](https://nodejs.org/en/)
+
+
+# Setup Project
+
+To get started, you need to create a project directory to house the files to be created in this guide.
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:25]:
```suggestion
```bash
```

File: docs/dapp/sdk/ether-js.md [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+mkdir ethers-sample
+cd ethers-sample
+```
+
+## Install ethers.js
+
+To install ethers.js run the following command in your terminal:
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:34]:
```suggestion
```bash
```

File: docs/dapp/sdk/ether-js.md [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+const url = 'RPC URL';
+const provider = new ethers.JsonRpcProvider(url)
+```
+
+# Reading data from the blockchain
+
+To read data from the blockchain, create a new `read.js` file in your project folder by running this command
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:61]:
```suggestion
```bash
```

File: docs/dapp/sdk/ether-js.md [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+touch read.js
+```
+
+After creating this file, initialize ethers as done in the `initialize` section. In this section, you ll learn how to read data from the blockchain e.g blockNumber, KLAY balance.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:65]:
```suggestion
After creating this file, initialize ethers as done in the `initialize` section. In this section, you will learn how to read data from the blockchain (e.g., blockNumber, KLAY balance).
```

File: docs/dapp/sdk/ether-js.md [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+async function getBlockNumber() {
+
+    const blocknumber  = await provider.getBlockNumber()
+    console.log("blocknumber", blocknumber)
+    
+}
+
+async function getKlayBalance() {
+
+    const klayBalance  = await provider.getBalance(“paste wallet address”)
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:80]:
```suggestion
    const klayBalance  = await provider.getBalance("paste wallet address")
```

File: docs/dapp/sdk/ether-js.md [hunk 21]
```diff
@@ -0,0 +1,234 @@
...
+To run the script and read data from the blockchain, you can run the following command in your terminal: 
+
+```js
+node read.js
+```
+
+If the transaction was succesful, you'll see the block number and user’s KLAY balance been logged in your terminal
+
+
+# Sending transaction to the blockchain
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:103]:
```suggestion
# Sending a transaction to the blockchain
```

File: docs/dapp/sdk/ether-js.md [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+
+To send transaction to the blockchain, create a new `send.js` file in your project folder by running this command:
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:107]:
```suggestion
```bash
```

File: docs/dapp/sdk/ether-js.md [hunk 23]
```diff
@@ -0,0 +1,234 @@
...
+node read.js
+```
+
+If the transaction was succesful, you'll see the block number and user’s KLAY balance been logged in your terminal
+
+
+# Sending transaction to the blockchain
+
+To send transaction to the blockchain, create a new `send.js` file in your project folder by running this command:
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:105]:
```suggestion
To send a transaction to the blockchain, create a new `send.js` file in your project folder by running this command:
```

File: docs/dapp/sdk/ether-js.md [hunk 24]
```diff
@@ -0,0 +1,234 @@
...
+
+```js
+touch send.js
+```
+
+After creating this file, initialize ethers as done in the `initialize` section. In this section, you ll learn how to send transaction to the blockchain e.g send KLAY to an address.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:111]:
```suggestion
After creating this file, initialize ethers as done in the `initialize` section. In this section, you will learn how to send a transaction to the blockchain (e.g., send KLAY to an address).
```

File: docs/dapp/sdk/ether-js.md [hunk 25]
```diff
@@ -0,0 +1,234 @@
...
+
+To see this in action, paste the following code in your `send.js`
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:113]:
```suggestion
To see this in action, paste the following code in your `send.js`.
```

File: docs/dapp/sdk/ether-js.md [hunk 26]
```diff
@@ -0,0 +1,234 @@
...
+
+```js
+const privKey = “Paste Privatekey”
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:116]: I think it could be good to add the code in the initialize section. In this case, the reader also should copy the code in two different sections.

File: docs/dapp/sdk/ether-js.md [hunk 27]
```diff
@@ -0,0 +1,234 @@
...
+const signer = new ethers.Wallet(privKey, provider)
+
+
+async function sendTx() {
+
+    const tx = await signer.sendTransaction({
+               to: "0x8B93b7d7Ed0960e9A0090851334e19b3c451E4E9",
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:123]: Is it okay to leave the address here?

File: docs/dapp/sdk/ether-js.md [hunk 28]
```diff
@@ -0,0 +1,234 @@
...
+sendTx();
+```
+
+**Output**
+
+To run the script and send data to the blockchain, you can run the following command in your terminal: 
+
+```js
+node send.js
+```
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:145]: it would be also good to add the result of the execution.

File: docs/dapp/sdk/ether-js.md [hunk 29]
```diff
@@ -0,0 +1,234 @@
...
+
+If the transaction was succesful, you'll see the transaction receipt been logged in your terminal.
+
+# Interact with smart contracts
+
+To interact with an existing smart contract on Klaytn, create a new `interact.js` file in your project folder by running this command:
+
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:154]:
```suggestion
```bash
```

File: docs/dapp/sdk/ether-js.md [hunk 30]
```diff
@@ -0,0 +1,234 @@
...
+touch interact.js
+```
+
+After creating this file, initialize ethers as done in the `initialize` section. In this section, you will use ethers.js to interact with a smart contract on Klaytn by instantiating a `Contract` object using the ABI and address of a deployed contract:
+
+For the purpose of this guide, a simple_storage contract was compiled and deployed on [Remix IDE](https://docs.klaytn.foundation/content/dapp/tutorials/connecting-remix). We will be sending transaction to the contract by calling the `store` function and also reading from it by calling the `retrieve` function.
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:160]:
```suggestion
For the purpose of this guide, a simple_storage contract was compiled and deployed on [Remix IDE](https://docs.klaytn.foundation/content/dapp/tutorials/connecting-remix). We will be sending a transaction to the contract by calling the `store` function and also reading from it by calling the `retrieve` function.
```

File: docs/dapp/sdk/ether-js.md [hunk 31]
```diff
@@ -0,0 +1,234 @@
...
+
+
+To see this in action, paste the following code in your `interact.js`
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:163]:
```suggestion
To see this in action, paste the following code in your `interact.js`.
```

File: docs/dapp/sdk/ether-js.md [hunk 32]
```diff
@@ -0,0 +1,234 @@
...
+
+```js
+// replace with your contract abi
```
Comment (kjhman21) [inline docs/dapp/sdk/ether-js.md:166]:
```suggestion
// replace with your contract ABI
```

**Comment by kjeom** [comment]
It's a great way to minimize the contents that describe how to use web3 SDK (environment setting and small example would be great like [web3j quick start](https://docs.web3j.io/4.8.7/quickstart/)
OpenSDK that will be developed and integrated with web3, give you more examples using Web3 SDK components.

File: docs/getting-started/hardhat/hardhat.md [hunk 33]
```diff
@@ -0,0 +1,412 @@
...
+* Paste this command in your terminal to create a .env file
+
+```js
+touch .env
+```
+
+* After creating our file, let's configure our .env file to look like this:
+
+```js
+ KLAYTN_BAOBAB_URL= “Your AllthatNode Baobab RPC link”
```
Comment (ian0371) [inline docs/getting-started/hardhat/hardhat.md:101]: curly quotes -> non-curly quotes (`""`)

File: docs/getting-started/hardhat/hardhat.md [hunk 34]
```diff
@@ -0,0 +1,412 @@
...
+ PRIVATE_KEY= “your private key copied from MetaMask wallet”
```
Comment (ian0371) [inline docs/getting-started/hardhat/hardhat.md:102]:
```suggestion
 KLAYTN_BAOBAB_URL= "Your AllthatNode Baobab RPC link"
 PRIVATE_KEY= "your private key copied from MetaMask wallet"
```

File: docs/getting-started/hardhat/hardhat.md [hunk 35]
```diff
@@ -0,0 +1,412 @@
...
+    console.log(`Deploying contracts with the account: ${deployer.address}`);
+    console.log(`Account balance: ${(await deployer.getBalance()).toString()}`);
+
+  const sbt = await ethers.getContractFactory("SoulBoundToken");
+  const sbtContract = await sbt.deploy();
+
+
+  await sbtContract.deployed();
+
+ console.log(`Token contract is deployed to ${sbtContract.address}`);
```
Comment (ian0371) [inline docs/getting-started/hardhat/hardhat.md:321]: Can we also print the KlaytnScope/KlaytnFinder URL so users can copy-paste in the browser?

File: docs/SUMMARY.md [hunk 36]
```diff
@@ -202,7 +203,7 @@
       * [subbridge](dapp/json-rpc/api-references/subbridge.md)
     * [Transaction Error Codes](dapp/json-rpc/transaction-error-codes.md)
   * [RPC Service Providers](dapp/rpc-service.md)
-  * [Klaytn SDKs](dapp/sdk/README.md)
+  * [Client SDK & Libraries](dapp/sdk/README.md)
```
Comment (kjeom) [inline docs/SUMMARY.md:206]: How about naming "SDK Libraries for interacting with Klaytn Node"

File: docs/getting-started/hardhat/hardhat.md (hunk 7 above)
Comment (ayo-klaytn) [reply docs/getting-started/hardhat/hardhat.md:174]: Will modify that to KIP17

File: docs/SUMMARY.md (hunk 8 above)
Comment (ayo-klaytn) [reply docs/SUMMARY.md:41]: yes trying to restructure the getting started section to include: Deploying smart contracts using Hardhat and  Klaytn Endpoint Node(KEN)

Comment (ayo-klaytn) [reply docs/SUMMARY.md:41]: yes trying to restructure the getting started section to include: Deploying smart contracts using Hardhat and  Klaytn Endpoint Node(KEN)

File: docs/dapp/sdk/ether-js.md (hunk 26 above)
Comment (ayo-klaytn) [reply docs/dapp/sdk/ether-js.md:115]: Got it

File: docs/dapp/sdk/ether-js.md (hunk 28 above)
Comment (ayo-klaytn) [reply docs/dapp/sdk/ether-js.md:144]: yeah right! A screenshot of the transaction receipt.

That's why I added the output of the tx below the command.

File: docs/SUMMARY.md (hunk 36 above)
Comment (ayo-klaytn) [reply docs/SUMMARY.md:206]: Yeah works fine. But also "Client SDK and Libraries" seem to be short.

@aidan-kwon what do you think?

Comment (ayo-klaytn) [reply docs/SUMMARY.md:206]: So something like "SDK & Libraries for interacting with Klaytn Node" should work?

Comment (kjeom) [reply docs/SUMMARY.md:206]: "SDK & Libraries for interacting with Klaytn Node" would be great.

File: docs/dapp/sdk/ether-js.md (hunk 28 above)
Comment (kjhman21) [reply docs/dapp/sdk/ether-js.md:151]: Where's the screenshot? It should be linked somewhere here with an image tag like:
```
![alt text](image/url)
```

Comment (ayo-klaytn) [reply docs/dapp/sdk/ether-js.md:151]: Added the screenshot. What I meant was the result of the tx can be explained in plain text as we have now.

File: docs/dapp/sdk/web3-js.md [hunk 37]
```diff
@@ -0,0 +1,255 @@
+![](./images/klaytn-web3js.png)
+
+# introduction
+
+[web3.js](https://web3js.readthedocs.io/) is a JavaScript library that allows developers to interact with EVM-compatible blockchain networks like Klaytn. With Klaytn supporting features for[ Ethereum Equivalence](https://medium.com/klaytn/using-ethereum-tools-in-klaytn-dc068d48de04), Ethereum tools such as web3.js can be used on Klaytn without any significant modifications.
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:5]:
```suggestion
[web3.js](https://web3js.readthedocs.io/) is a JavaScript library that allows developers to interact with EVM-compatible blockchain networks like Klaytn. With Klaytn supporting features for[Ethereum Equivalence](https://medium.com/klaytn/using-ethereum-tools-in-klaytn-dc068d48de04), Ethereum tools such as web3.js can be used on Klaytn without any significant modifications.
```

File: docs/dapp/sdk/web3-js.md [hunk 38]
```diff
@@ -0,0 +1,255 @@
...
+* RPC Endpoint: you can get this from one of the supported [Endpoint Providers](https://docs.klaytn.foundation/content/dapp/json-rpc/public-en).
+* Test KLAY from [Faucet](https://baobab.wallet.klaytn.foundation/faucet): fund your account with sufficient KLAY.
+* [NodeJS and NPM](https://nodejs.org/en/)
+
+
+# Setup Project
+
+To get started, you need to create a project directory to house the files to be created in this guide.
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:25]:
```suggestion
```bash
```

File: docs/dapp/sdk/web3-js.md [hunk 39]
```diff
@@ -0,0 +1,255 @@
...
+mkdir web3js-sample
+cd web3js-sample
+```
+
+## Install web3.js
+
+To install web3.js run the following command in your terminal:
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:34]:
```suggestion
```bash
```

File: docs/dapp/sdk/web3-js.md [hunk 40]
```diff
@@ -0,0 +1,255 @@
...
+const privateKey = "Paste private key";
+```
+
+# Reading data from the blockchain
+
+To read data from the blockchain, create a new `read.js` file in your project folder by running this command
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:65]:
```suggestion
```bash
```

File: docs/dapp/sdk/web3-js.md [hunk 41]
```diff
@@ -0,0 +1,255 @@
...
+touch read.js
+```
+
+After creating this file, initialize `web3` as done in the `initialize` section. In this section, you ll learn how to read data from the blockchain e.g blockNumber, KLAY balance.
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:69]:
```suggestion
After creating this file, initialize `web3` as done in the `initialize` section. In this section, you'll learn how to read data from the blockchain (e.g., blockNumber, KLAY balance).
```

File: docs/dapp/sdk/web3-js.md [hunk 42]
```diff
@@ -0,0 +1,255 @@
...
+
+To read data from the blockchain, create a new `read.js` file in your project folder by running this command
+
+```js
+touch read.js
+```
+
+After creating this file, initialize `web3` as done in the `initialize` section. In this section, you ll learn how to read data from the blockchain e.g blockNumber, KLAY balance. 
+
+To see this in action, paste the following code in your `read.js`
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:71]:
```suggestion
To see this in action, paste the following code in your `read.js`.
```

File: docs/dapp/sdk/web3-js.md [hunk 43]
```diff
@@ -0,0 +1,255 @@
...
+getLatestBlock();
+getKlayBalance();
+
+```
+
+**Output**
+
+To run the script and read data from the blockchain, you can run the following command in your terminal: 
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:95]:
```suggestion
```bash
```

File: docs/dapp/sdk/web3-js.md [hunk 44]
```diff
@@ -0,0 +1,255 @@
...
+```
+Further, you need to add your private key to sign transactions. Add the code below to the existing code:
+
+```js
+const privateKey = "Paste private key";
+```
+
+# Reading data from the blockchain
+
+To read data from the blockchain, create a new `read.js` file in your project folder by running this command
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:63]:
```suggestion
To read data from the blockchain, create a new `read.js` file in your project folder by running this command:
```

File: docs/dapp/sdk/web3-js.md [hunk 45]
```diff
@@ -0,0 +1,255 @@
...
+node read.js
+```
+
+If the transaction was succesful, you'll see the block number and user’s KLAY balance been logged in your terminal.
+
+# Sending transaction to the blockchain
+
+To send transaction to the blockchain, create a new `send.js` file in your project folder by running this command:
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:105]:
```suggestion
```bash
```

File: docs/dapp/sdk/web3-js.md [hunk 46]
```diff
@@ -0,0 +1,255 @@
...
+touch send.js
+```
+
+
+After creating this file, initialize `web3` as done in the `initialize` section. In this section, you ll learn how to send transaction to the blockchain e.g send KLAY to an address.
+
+To see this in action, paste the following code in your `send.js`
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:112]:
```suggestion
To see this in action, paste the following code in your `send.js`.
```

File: docs/dapp/sdk/web3-js.md [hunk 47]
```diff
@@ -0,0 +1,255 @@
...
+
+If the transaction was succesful, you'll see the transaction receipt been logged in your terminal.
+
+![](images/send-web3.png)
+
+# Interact with smart contracts
+
+To interact with an existing smart contract on Klaytn, create a new `interact.js` file in your project folder by running this command:
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:154]:
```suggestion
```bash
```

**Comment by kjhman21** [comment]
What do you think if we provide a github repository for the examples? It could be much easier to execute the examples here.

File: docs/dapp/sdk/web3-js.md [hunk 48]
```diff
@@ -0,0 +1,255 @@
...
+touch interact.js
+```
+
+After creating this file, initialize `web3` as done in the `initialize` section. In this section, you will use web3.js to interact with a smart contract on Klaytn by instantiating a `Contract` object using the ABI and address of a deployed contract:
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:158]:
```suggestion
After creating this file, initialize `web3` as done in the `initialize` section. In this section, you will use web3.js to interact with a smart contract on Klaytn by instantiating a `Contract` object using the ABI and address of a deployed contract.
```

File: docs/dapp/sdk/web3-js.md [hunk 49]
```diff
@@ -0,0 +1,255 @@
...
+
+For the purpose of this guide, a simple_storage contract was compiled and deployed on [Remix IDE](https://docs.klaytn.foundation/content/dapp/tutorials/connecting-remix). We will be sending transaction to the contract by calling the `store` function and also reading from it by calling the `retrieve` function.
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:160]:
```suggestion
For the purpose of this guide, a simple_storage contract was compiled and deployed on [Remix IDE](https://docs.klaytn.foundation/content/dapp/tutorials/connecting-remix). We will be sending a transaction to the contract by calling the `store` function and also reading from it by calling the `retrieve` function.
```

File: docs/dapp/sdk/web3-js.md [hunk 50]
```diff
@@ -0,0 +1,255 @@
...
+
+
+To see this in action, paste the following code in your `interact.js`
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:163]:
```suggestion
To see this in action, paste the following code in your `interact.js`.
```

File: docs/dapp/sdk/web3-js.md [hunk 51]
```diff
@@ -0,0 +1,255 @@
...
+    const contractAddress = "0x472a1226796b6a0918DC78d40b87d750881fdbDC"
+    
+    const contract = new web3.eth.Contract(abi, contractAddress);
+    
+    // Can replace `10` with any value you want to store
+    const storeTx = contract.methods.store(10);
+    
+   // send transaction to smart contract
+   // modify contract
+    async function setValue() {
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:213]:
```suggestion
    // send transaction to smart contract
    // modify contract
    async function setValue() {
```

It would be good to lint the source code. Indentation should be refined.

File: docs/dapp/sdk/web3-js.md [hunk 52]
```diff
@@ -0,0 +1,255 @@
...
+      setValue();
+      retrieveValue();
+
+```
+
+**Output**
+
+To run the script and interact with smart contracts, you can run the following command in your terminal: 
+
+```js
```
Comment (kjhman21) [inline docs/dapp/sdk/web3-js.md:248]:
```suggestion
```bash
```

**Comment by ayo-klaytn** [comment]
> What do you think if we provide a github repository for the examples? It could be much easier to execute the examples here.

Awesome! That works, the repo houses all sample code. Could also be in a gist file.

**Comment by kjhman21** [comment]
@ayo-klaytn Yes, and make a link in the docs. it would be also good to have a repository in Klaytn org instead of your own repo.

**Comment by ayo-klaytn** [comment]
> @ayo-klaytn Yes, and make a link in the docs. it would be also good to have a repository in Klaytn org instead of your own repo.

@kjhman21 Yeah sure, will definitely add the link. Can you help with the creation of this repo and share ser?

File: docs/SUMMARY.md [hunk 53]
```diff
@@ -203,7 +204,7 @@
       * [subbridge](dapp/json-rpc/api-references/subbridge.md)
     * [Transaction Error Codes](dapp/json-rpc/transaction-error-codes.md)
   * [RPC Service Providers](dapp/rpc-service.md)
-  * [Klaytn SDKs](dapp/sdk/README.md)
+  * [SDK & Libraries for interacting with Klaytn Node](dapp/sdk/README.md)
```
Comment (toniya-klaytn) [inline docs/SUMMARY.md:207]:
```suggestion
  * [SDK & Libraries](dapp/sdk/README.md)
```

Comment (ayo-klaytn) [reply docs/SUMMARY.md:207]: @toniya-klaytn we already had a conversation about the title to use for this section. Do you mind checking it out?

- [Aidan's suggestion](https://github.com/klaytn/klaytn-docs/pull/545/files/ac8b356edf3b6037071788f78168835da560770d)
- [Kjeom's suggestion](https://github.com/klaytn/klaytn-docs/pull/545#discussion_r1126147805)

So we can finalize which title to use. Thanks

Comment (toniya-klaytn) [reply docs/SUMMARY.md:207]: oops sorry, we can go ahead with the suggestions then. Thanks

**Review by toniya-klaytn — APPROVED** [review]

**Review by kjhman21 — APPROVED** [review]
LGTM. Thanks for your hard work!

**Review by kjeom — APPROVED** [review]
